pkg/resources/database/datasource_uri.go

```diff
@@ -283,9 +281,10 @@ func (d *DataSourceURI) Read(ctx context.Context, req datasource.ReadRequest, re
 			return
 		}
 
+		params = res.URIParams
 		data.Schema = types.StringValue("mysql")
 
-		creds, err := client.RevealDBAASMysqlUserPassword(ctx, data.Name.ValueString(), adminUsername)
+		creds, err := client.RevealDBAASMysqlUserPassword(ctx, data.Name.ValueString(), params["user"].(string))
```

Comment:
Can you implement a check that `params` map has `user` key and that value is a string?
Although it's always expected to exist, in case of API hickup it will cause a panic in provider.